common/testing/taskpoller/taskpoller.go

```diff
@@ -84,70 +96,139 @@ func New(
+func (p *TaskPoller) PollAndHandleActivityTask(
+	tv *testvars.TestVars,
+	handler func(task *workflowservice.PollActivityTaskQueueResponse) (*workflowservice.RespondActivityTaskCompletedRequest, error),
+	funcs ...optionFunc,
+) (*workflowservice.RespondActivityTaskCompletedResponse, error) {
+	return p.
+		PollActivityTask(&workflowservice.PollActivityTaskQueueRequest{}).
+		HandleTask(tv, handler, funcs...)
+}
+
+// HandleActivityTask invokes the provided handler with the provided task, and completes/fails the task accordingly.
+// Any unspecified but required request and response fields are automatically generated using `tv`.
+// Returning an error from `handler` fails the task.
+func (p *TaskPoller) HandleActivityTask(
+	tv *testvars.TestVars,
+	task *workflowservice.PollActivityTaskQueueResponse,
+	handler func(task *workflowservice.PollActivityTaskQueueResponse) (*workflowservice.RespondActivityTaskCompletedRequest, error),
+	funcs ...optionFunc,
+) (*workflowservice.RespondActivityTaskCompletedResponse, error) {
+	p.t.Helper()
+	options := newOptions(tv, funcs)
+	ctx, cancel := newContext(options)
+	defer cancel()
+	ap := activityTaskPoller{TaskPoller: p}
+	return ap.handleTask(ctx, options, task, handler)
+}
+
+// HandleTask invokes the provided handler with the task poll result, and completes/fails the task accordingly.
+// Any unspecified but required request and response fields are automatically generated using `tv`.
+// Returning an error from `handler` fails the task.
+// If no task is available, it returns NoTaskAvailable.
+func (p *activityTaskPoller) HandleTask(
+	tv *testvars.TestVars,
+	handler func(task *workflowservice.PollActivityTaskQueueResponse) (*workflowservice.RespondActivityTaskCompletedRequest, error),
+	funcs ...optionFunc,
+) (*workflowservice.RespondActivityTaskCompletedResponse, error) {
+	p.t.Helper()
+	options := newOptions(tv, funcs)
+	ctx, cancel := newContext(options)
+	defer cancel()
+	return p.pollAndHandleTask(ctx, options, handler)
 }
 
 //revive:disable-next-line:cognitive-complexity
-func (p *TaskPoller) pollWorkflowTask(
+func (p *workflowTaskPoller) pollTask(
 	ctx context.Context,
-	opts *Options,
+	opts *options,
 ) (*workflowservice.PollWorkflowTaskQueueResponse, error) {
 	p.t.Helper()
-	taskQueue := opts.tv.TaskQueue()
-	if opts.pollStickyTaskQueue {
-		taskQueue = opts.tv.StickyTaskQueue()
-	}
 
-	resp, err := p.client.PollWorkflowTaskQueue(
-		ctx,
-		&workflowservice.PollWorkflowTaskQueueRequest{
-			Namespace: p.namespace,
-			TaskQueue: taskQueue,
-			Identity:  opts.tv.WorkerIdentity(),
-		})
+	req := common.CloneProto(p.pollWorkflowTaskRequest)
+	if req.Namespace == "" {
+		req.Namespace = p.namespace
+	}
+	if req.TaskQueue == nil {
+		req.TaskQueue = opts.tv.TaskQueue()
+	}
+	if req.Identity == "" {
+		req.Identity = opts.tv.WorkerIdentity()
+	}
+	resp, err := p.client.PollWorkflowTaskQueue(ctx, req)
 	if err != nil {
 		return nil, err
 	}
 	if resp == nil || len(resp.TaskToken) == 0 {
```

Comment:
This is probably some legacy but I don't know when `TaskToken` can be 0 length. Let's remove this condition and see if it breaks anything.